File: app/src/main/java/com/gtm/pustak/base/BaseActivity.java
package com.gtm.pustak.base;

import android.os.Bundle;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import com.gtm.archcomponents.R;

import io.reactivex.disposables.CompositeDisposable;

public abstract class BaseActivity extends AppCompatActivity {

    public Toolbar toolbar;
    public CompositeDisposable compositeDisposable;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutResource());
        toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        compositeDisposable = new CompositeDisposable();
    }

    public abstract int getLayoutResource();

    public void showToast(String message) {

        Toast.makeText(this, message, Toast.LENGTH_SHORT).show();
    }

    @Override
    protected void onPause() {
        super.onPause();
        //todo clear observable disposable here.
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        compositeDisposable.clear();
        compositeDisposable.dispose();

    }
}

File: app/src/main/java/com/gtm/pustak/kotlin/base/BaseActivity.kt
package com.gtm.pustak.kotlin.base

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.gtm.pustak.R
import io.reactivex.disposables.CompositeDisposable

abstract class BaseActivity<B> : AppCompatActivity() {


    var toolbar: Toolbar? = null
    var binding: B? = null
    lateinit var compositeDisposable: CompositeDisposable
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = getLayoutResource()
        toolbar = findViewById(R.id.toolbar)
        setSupportActionBar(toolbar)
        compositeDisposable = CompositeDisposable()
    }

    abstract fun getLayoutResource(): B

    fun showToast(message: String) {
        Toast.makeText(this, message, Toast.LENGTH_SHORT).show()
    }

    override fun onPause() {
        super.onPause()
    }

    override fun onDestroy() {
        super.onDestroy()
        compositeDisposable.clear()
        compositeDisposable.dispose()
    }

}
